samples/excel/89-preview-apis/protect-data-in-worksheet-and-workbook-structure.yaml

```diff
@@ -0,0 +1,258 @@
+id: excel-protect-data-in-worksheet-and-workbook-structure
+name: Protect data in worksheet and the workbook structure
+description: Shows how to protect data in a worksheet and the workbook structure.
+host: EXCEL
+api_set:
+    ExcelApi: 1.7
+script:
+    content: |
+        $("#setup").click(() => tryCatch(setup));
```

Comment:
It looks like you've got a somewhat complex structure in order to avoid repetitive code. This is going to be problematic when we try to extract these functions for use in the reference topics. For example, your helper methods would have to be used for _both_ the `protect` and `unprotect` APIs, and the reader will have to figure out that the parameters passed to the method is what determines what it's going to do. 
Our Script Lab snippets should be relentlessly focused on showing clearly the Office APIs. All other considerations, including reusing code, are relatively unimportant. Devs get their best practice education elsewhere.